client/homebrew/utils/versionHistory.js

```diff
@@ -0,0 +1,120 @@
+export const HISTORY_PREFIX = 'HOMEBREWERY-HISTORY'
+// const HISTORY_SAVE_DELAYS = {
+//  0: 0,           // 0 minutes (if not specified)
+// 	1: 2,			// 2 minutes
+// 	2: 10,			// 10 minutes
+// 	3: 60,			// 60 minutes
+// 	4: 12 * 60,     // 12 hours
+// 	5: 2 * 24 * 60  // 2 days
+// };
+// 
+// const GARBAGE_COLLECT_DELAY = 28 * 24 * 60; // 28 days
+
+
+// <=================== TEST VALUES STARTS ===================>
+
+// Test values
+const HISTORY_SAVE_DELAYS = {
+    0: 0,           // 0 minutes (if not specified)
+	1: 1,			// 1 minutes
+	2: 2,			// 2 minutes
+	3: 3,			// 3 minutes
+	4: 4,           // 4 minutes
+	5: 5            // 5 minutes
+};
+const GARBAGE_COLLECT_DELAY = 10; // 10 minutes
+
+// <==================== TEST VALUES ENDS ====================>
+
+
+
+const DEFAULT_STORED_BREW = {
+    shareId  : 'default_brew',
+    expireAt : '2000-01-01T00:00:00.000Z'
+};
+
+function getKeyBySlot(brew, slot){
+    return `${HISTORY_PREFIX}-${brew.shareId}-${slot}`;
+};
+
+function getVersionBySlot(brew, slot){
+    // Read stored brew data
+    // - If it exists, parse data to object
+    // - If it doesn't exist, pass default object
+    const key = getKeyBySlot(brew, slot);
+    const storedVersion = localStorage.getItem(key);
+    const output = storedVersion ? JSON.parse(storedVersion) : { ...DEFAULT_STORED_BREW, ...brew };
+    return output;
+};
+
+function updateStoredBrew(brew, slot=0){
+    const archiveBrew = {};
+
+    // Data from brew to be stored
+    archiveBrew.title = brew.title;
+    archiveBrew.text = brew.text;
+    archiveBrew.style = brew.style;
+    archiveBrew.version = brew.version;
+    archiveBrew.shareId = brew.shareId;
+
+    // Calculated values
+    archiveBrew.savedAt = new Date();
```

Comment:
It looks like `savedAt` isn't used anymore.

Reply:
Correct, I was using it for garbage collection and then right before commit, I changed GC to use `expireAt` instead. Still not sure which will end up being the correct approach.

Reply:
I suspect it will still be useful to keep the `savedAt` data for the UI, although given the current functionality, it might be more useful if it is not overwritten during the storage slot movement process - so the UI can indicate the age of the data be restored, e.g. "Saved data from 19:00 August 10, 2024"

Reply:
I've changed `savedAt` to no longer be overwritten during the updating process, so the time that the stored data was current will be available to the UI.